x/interchainstaking/keeper/delegation_test.go

```diff
@@ -611,65 +613,298 @@ func (s *KeeperTestSuite) TestStoreGetDeleteDelegation() {
+		_, found = qApp.InterchainstakingKeeper.GetDelegation(ctx, &zone, delegator.String(), validator.String())
 		s.Require().False(found)
 
 		newDelegation := types.NewDelegation(delegator.String(), validator.String(), sdk.NewCoin("uatom", sdk.NewInt(5000)))
-		app.InterchainstakingKeeper.SetDelegation(ctx, &zone, newDelegation)
+		qApp.InterchainstakingKeeper.SetDelegation(ctx, &zone, newDelegation)
 
-		fetchedDelegation, found := app.InterchainstakingKeeper.GetDelegation(ctx, &zone, delegator.String(), validator.String())
+		fetchedDelegation, found := qApp.InterchainstakingKeeper.GetDelegation(ctx, &zone, delegator.String(), validator.String())
 		s.Require().True(found)
 		s.Require().Equal(newDelegation, fetchedDelegation)
 
-		allDelegations := app.InterchainstakingKeeper.GetAllDelegations(ctx, &zone)
+		allDelegations := qApp.InterchainstakingKeeper.GetAllDelegations(ctx, &zone)
 		s.Require().Len(allDelegations, 1)
 
-		err := app.InterchainstakingKeeper.RemoveDelegation(ctx, &zone, newDelegation)
+		err := qApp.InterchainstakingKeeper.RemoveDelegation(ctx, &zone, newDelegation)
 		s.Require().NoError(err)
 
-		allDelegations2 := app.InterchainstakingKeeper.GetAllDelegations(ctx, &zone)
+		allDelegations2 := qApp.InterchainstakingKeeper.GetAllDelegations(ctx, &zone)
 		s.Require().Len(allDelegations2, 0)
 	})
 }
 
-func (s *KeeperTestSuite) TestFlushDelegations() {
-	s.Run("flush delegations", func() {
-		s.SetupTest()
-		s.setupTestZones()
-
-		app := s.GetQuicksilverApp(s.chainA)
-		ctx := s.chainA.GetContext()
-
-		t := time.Now()
-		receipt := types.Receipt{
-			ChainId: s.chainB.ChainID,
-			Sender:  utils.GenerateAccAddressForTest().String(),
-			Txhash:  "6F2EEAE407E620C3D8F68C535C899CD7F1BAB1680686DF41C2FC38D139B940E9",
-			Amount: sdk.NewCoins(
-				sdk.NewCoin(
-					"ujuno",
-					sdk.NewIntFromUint64(2000000),
-				),
-			),
-			FirstSeen: &t,
-		}
+func (s *KeeperTestSuite) TestFlushOutstandingDelegations() {
+	userAddress := utils.GenerateAccAddressForTest().String()
+	denom := "uatom"
+	tests := []struct {
+		name               string
+		setStatements      func(ctx sdk.Context, quicksilver *app.Quicksilver)
+		delAddrBalance     sdk.Coin
+		mockAck            bool
+		expectedDelegation sdk.Coins
+		assertStatements   func(ctx sdk.Context, quicksilver *app.Quicksilver) bool
+	}{
+		{
+			name:           "zero pending delegations, no pending receipts, no exclusion receipts",
+			setStatements:  func(ctx sdk.Context, quicksilver *app.Quicksilver) {},
+			delAddrBalance: sdk.NewCoin("uatom", sdkmath.ZeroInt()),
+			assertStatements: func(ctx sdk.Context, quicksilver *app.Quicksilver) bool {
+				return true
+			},
+		},
+		{
+			name: "zero pending delegations, 2 pending receipts and no exclusion receipts",
+			setStatements: func(ctx sdk.Context, quicksilver *app.Quicksilver) {
+				cutOffTime := ctx.BlockTime().AddDate(0, 0, -1)
+				pendingRecieptTime := cutOffTime.Add(-2 * time.Hour)
```

Comment:
can we make these `rcpt1time` and `rcpt2time` because they are _not_ excluded here - confusing to read and reason about.